### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by araspaas**
Refactor Api calls into POROS(Plain old ruby objects), movie service, and movie facade. and add tests for latter classes.

- Start to refactor using poros, services, and facades with movies
- Update movie index view to use PORO method instead of hash/key calls
- Update movie show view with PORO methods instead of hash/key calls
- Create top 40 and movie data methods
- Update movie details PORO to call on MovieService methods
- Update movie facade with top 40 and movie data methods
- Update index and show action to call on movie facade for movie data
- Update spec to record new VCR cassette episodes
- Update review and cast to call PORO methods instead of hash/key calls
- Create new POROs for actor and review
- Add methods for cast and review API lookups
- Update MovieFacade to call MovieService methods
- Update movies controller show action to call on movie facade
- Add movie_service spec
- Add tests for the movie service class
- Add tests for review PORO
- Add tests for the MovieDetails PORO
- Add tests for Actor PORO
- Add test for when a user tries to log in and is already logged in
- Remove redundant tests and add sad path test for creating a bad party
- Add tests for the movie facade 
- Update sessions controller to actually hit a sad path test
- Update party controller to hit a sad path when creating a new party
- Fix rubocop offenses

### CODE REVIEW

File: app/services/movie_service.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  def self.review_data(movie_id)
+    results = conn.get("/3/movie/#{movie_id}/reviews?")
+    parse(results)
+  end
+
+  def self.conn
+    Faraday.new(url: 'https://api.themoviedb.org') do |f|
+      f.params[:api_key] = ENV['MOVIES_API_KEY']
+      # f.headers["include_adult"] = "false"
```
Comment (smcunning) [inline app/services/movie_service.rb]: This is another refactor I'd like to take a stab at - currently we're not guarding against 'adult' movies.

File: spec/features/parties/new_spec.rb [hunk 2]
```diff
@@ -67,30 +53,41 @@
...
       movie = Movie.last
       within ".party_form" do
         fill_in :duration, with: movie.duration
         fill_in :date, with: "12-08-2020"
         fill_in :start_time, with: "7:00"
+        click_button "Create Viewing Party"
+      end
+    end
+
+    it "text" do
```
Comment (smcunning) [inline spec/features/parties/new_spec.rb]: We need to update this test name to be specific

File: spec/features/welcome_spec.rb [hunk 3]
```diff
@@ -52,5 +52,24 @@
       expect(current_path).to eq(dashboard_index_path)
       expect(page).to have_content("The page you were looking for doesn't exist")
     end
+
+    it "text" do
```
Comment (smcunning) [inline spec/features/welcome_spec.rb]: Another one to update the test name on

File: app/controllers/discover_controller.rb [hunk 4]
```diff
@@ -1,4 +1,3 @@
 class DiscoverController < ApplicationController
-  def index
-  end
+  def index; end
```
Comment (KieraAllen) [inline app/controllers/discover_controller.rb:2]: Thank you Rubocop!

File: app/controllers/friendships_controller.rb [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
       flash[:error] = "I'm sorry your friend cannot be found"
     elsif friend.id == user_id
-      flash[:error] = "You can not add yourself as a friend"
+      flash[:error] = 'You can not add yourself as a friend'
     elsif friend_exists?(friend)
-      flash[:error] = "You have already added this friend"
-    else friend
-      Friendship.create({user_id: current_user.id, friend_id: friend.id})
+      flash[:error] = 'You have already added this friend'
+    else
```
Comment (KieraAllen) [inline app/controllers/friendships_controller.rb:16]: Now I understand why we were getting the error for `friend`... it wasn't necessary after the `else`

File: app/controllers/movies_controller.rb [hunk 6]
```diff
@@ -1,32 +1,21 @@
...
+      redirect_to '/discover'
+    elsif search_term
+      @movies = MovieFacade.movies_by_search(search_term)
     else
-      top_20 = Faraday.get("https://api.themoviedb.org/3/movie/top_rated?api_key=#{ENV['MOVIES_API_KEY']}&language=en-US&page=1")
-      second_20 = Faraday.get("https://api.themoviedb.org/3/movie/top_rated?api_key=#{ENV['MOVIES_API_KEY']}&language=en-US&page=2")
-      parsed_top_20 = JSON.parse(top_20.body, symbolize_names: :true)
-      parsed_second_20 = JSON.parse(second_20.body, symbolize_names: :true)
-      @movies = (parsed_top_20[:results] << parsed_second_20[:results]).flatten
+      @movies = MovieFacade.top_40_movies
```
Comment (KieraAllen) [inline app/controllers/movies_controller.rb:10]: Look at all that clean white space! (I'm viewing the changes side by side and there's a lot of extra space showing that the method's a lot cleaner now)

File: app/services/movie_service.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+  def self.review_data(movie_id)
+    results = conn.get("/3/movie/#{movie_id}/reviews?")
+    parse(results)
+  end
+
+  def self.conn
+    Faraday.new(url: 'https://api.themoviedb.org') do |f|
+      f.params[:api_key] = ENV['MOVIES_API_KEY']
+      f.params[:include_adult] = "false"
```
Comment (KieraAllen) [inline app/services/movie_service.rb:42]: Nice add @smcunning ! Wouldn't want any surprised ;)

File: spec/facades/movie_facade_spec.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
+require "rails_helper"
+
+describe "Movie Facade" do
+  it "returns a list of searched movies objects" do
+    VCR.use_cassette('movies_by_search', :record => :new_episodes) do
```
Comment (KieraAllen) [inline spec/facades/movie_facade_spec.rb:5]: Just for posterity... that `:record => :new_episodes` addition is so the VCR records new information when it's run again

File: spec/poros/actor_spec.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
+require "rails_helper"
```
Comment (KieraAllen) [inline spec/poros/actor_spec.rb:1]: The cutest PORO in all of creation

**Review by KieraAllen — APPROVED** [review]

**Pull request merged**